Sources/SwiftFormats/FormatStyle+CoreGraphics.swift

```diff
@@ -44,6 +44,32 @@ public struct CGPointParseStrategy: ParseStrategy {
 
 // MARK: -
 
+public extension CGPoint {
+    init(_ string: String) throws {
+        self = try CGPointFormatStyle().parseStrategy.parse(string)
+    }
+
+    init<T, Value>(_ value: Value, format: T) throws where T: ParseableFormatStyle, Value: StringProtocol, T.Strategy == CGPointParseStrategy {
```

Comment:
Neither this one that accepts the format, nor the next one that accepts the parse strategy are really needed for the CGPointFormatStyle, but I wanted to show off these two ways that could be used across the module to allow for customization of the parsing strategy.

Since Apple's official implementations for these kinds of convenience initializers are… inconsistent (that a polite way of saying it?). I've based these ones on the URL parsing functionality that was recently accepted through the [Swift Working Group by Charles Hu](https://forums.swift.org/t/url-formatstyle-and-parsestrategy/56607/5).

Reply:
So I've been wondering if there's some benefit to a `Parseable` (and `Formattable`) protocols that types can opt into with a default implementation covering most use cases. Obviously if Apple adds those protocols later we'd need to deprecate them here. But they seem very useful to get rid a lot of boilerplate and let the consume work with generic types... Any thoughts?

Reply:
I think there's a real benefit to define a protocol to represent "simple" types like this one, where the amount of customization for the output/input is nonexistent. I might play around with that next.